PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            _pager.Write(content, regexPatternToScrollTo);
+        }
+
+        private void WriteDynamicHelpContent(string commandName, string parameterName, bool isFullHelp)
+        {
+            var helpContent = _mockableMethods.GetDynamicHelpContent(commandName, parameterName, isFullHelp);
+
+            if (helpContent is string fullHelp && fullHelp.Length > 0)
+            {
+                string regexPatternToScrollTo = null;
+
+                if (!string.IsNullOrEmpty(parameterName))
+                {
+                    string upper = parameterName[0].ToString().ToUpperInvariant();
+                    string lower = parameterName[0].ToString().ToLowerInvariant();
+                    string remainingString = parameterName.Substring(1);
+                    regexPatternToScrollTo = $"-[{upper}|{lower}]{remainingString} [<|\\[]";
+                }
+
+                _mockableMethods.RenderFullHelp(fullHelp, regexPatternToScrollTo);
+            }
+            else if (helpContent is PSObject paramHelp)
+            {
+                WriteParameterHelp(paramHelp);
+            }
+        }
+
+        private void DynamicHelpImpl(bool isFullHelp)
+        {
+            if (isFullHelp)
+            {
+                _pager ??= new Pager();
+            }
+
+            int cursor = _singleton._current;
+            string commandName = null;
+            string parameterName = null;
+
+            foreach(var token in _singleton._tokens)
+            {
+                var extent = token.Extent;
+
+                if (extent.EndOffset > cursor)
+                {
+                    break;
+                }
+
+                if (token.TokenFlags == TokenFlags.CommandName)
+                {
+                    commandName = token.Text;
+                }
+
+                if (extent.StartOffset <= cursor && extent.EndOffset >= cursor)
+                {
+                    if (token.Kind == TokenKind.Parameter)
+                    {
+                        parameterName = ((ParameterToken)token).ParameterName;
+                        break;
+                    }
+                }
+            }
+
+            WriteDynamicHelpContent(commandName, parameterName, isFullHelp);
+        }
+
+        private void WriteDynamicHelpBlock(Collection<string> helpBlock)
+        {
+            var bufferWidth = _console.BufferWidth;
+
+            var dynHelp = new MultilineDisplayBlock
+            {
+                Singleton = this,
+                ItemsToDisplay = helpBlock
+            };
+
+            dynHelp.DrawMultilineBlock();
+            ReadKey();
+            dynHelp.Clear();
+        }
+
+        private void WriteParameterHelp(dynamic helpContent)
+        {
+            Collection<string> helpBlock;
+
+            if (helpContent == null || string.IsNullOrEmpty(helpContent?.Description?[0]?.Text))
```

Comment:
```suggestion
            if (string.IsNullOrEmpty(helpContent?.Description?[0]?.Text))
```
No need the null check as you are already using `?.`

Reply:
Fixed